core/client-entry.ts

```diff
@@ -116,6 +110,11 @@ const invokeClientEntry = async () => {
         }
       }))
     })
+    // if route is dispatched and there is no pending in router history
+    // if there is stil pending then 'beforeResolve' will handle app mount, otherwise 'beforeResolve' will not trigger
+    if (RouterManager.isRouteDispatched() && !(router as any).history.pending) {
+      app.$mount('#app')
+    }
```

Comment:
I guess it would be better to mix this edge case handler with the main condition (line 75) into single one. It will be much easier to understand how it exactly works

Reply:
I made update :) Also moved if in onReady callback on top of it.